policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/student_loan_interest_ald.py

```diff
@@ -0,0 +1,76 @@
+from policyengine_us.model_api import *
+
+
+class student_loan_interest_ald(Variable):
+    value_type = float
+    entity = Person
+    label = "Student loan interest ALD"
+    unit = USD
+    definition_period = YEAR
+    documentation = "Above-the-line deduction for student loan interest"
+    reference = "https://www.law.cornell.edu/uscode/text/26/221"
+
+    def formula(person, period, parameters):
+        interest = person("student_loan_interest", period)
+        p = parameters(period).gov.irs.ald.student_loan_interest
+        filing_status = person.tax_unit("filing_status", period)
+        capped_interest = min_(interest, p.cap[filing_status])
+        # The reduction is based on the taxpayer's modified adjusted gross income (MAGI)
+        # which is defined as AGI less ALDs
+        p_irs = parameters(period).gov.irs
+        not_dependent = ~person("is_tax_unit_dependent", period)
+        gross_income_sources = p_irs.gross_income.sources
+        # Unemployment compensation is not considered in MAGI
```

Comment:
Is this a MAGI definition specifically for this program? If so could you create a new variable for it, and a list parameter with excluded sources, with a legal code reference?

Reply:
It is specifically defined for the program, I have added the reference and paragraph to the var